tools/csv-writer/main.go

```diff
@@ -0,0 +1,827 @@
+)
+
+// Command-line parameters
+var (
+	showFile           = flag.Bool("showFile", false, "List all files in the S3 directory")
+	deleteFile         = flag.String("deleteFile", "", "Delete a specific file from S3")
+	glanceFile         = flag.String("glanceFile", "", "Glance the first 1024*1024 character of a specific file from S3")
+	fetchFile          = flag.String("fetchFile", "", "Fetch a specific file from S3, need to specify the local path.")
+	deleteFileByPrefix = flag.String("deleteFileByPrefix", "", "Delete all files with the specific prefix")
+	deleteAfterGen     = flag.Bool("deleteAfterGen", false, "Delete all files after generating (TEST ONLY!)")
+
+	tableInfo           = flag.String("tableInfo", "/home/admin/table_info.csv", "Path to table information")
+	localPath           = flag.String("localPath", "", "Local path to write file")
+	fileName            = flag.String("fileName", "testCSVWriter", "Base file name")
+	fileNameSuffixStart = flag.Int("fileNameSuffixStart", 0, "Start of file name suffix")
+	credentialPath      = flag.String("credential", "", "Path to S3 credential file")
+	rowNumPerFile       = flag.Int("rowNumPerFile", 10, "Number of rows to generate in each csv file")
+	pkBegin             = flag.Int("pkBegin", 0, "Begin of primary key, [begin, end)")
+	pkEnd               = flag.Int("pkEnd", 10, "End of primary key [begin, end)")
+	base64Encode        = flag.Bool("base64Encode", false, "Base64 encode the CSV file")
+	generatorNum        = flag.Int("generatorNum", 1, "Number of generator goroutines")
+	writerNum           = flag.Int("writerNum", 8, "Number of writer goroutines")
+
+	s3Path      = flag.String("s3Path", "gcs://global-sort-dir/testGenerateCSV", "S3 path")
+	s3AccessKey = flag.String("s3AccessKey", "", "S3 access key")
+	s3SecretKey = flag.String("s3SecretKey", "", "S3 secret key")
+	s3Region    = flag.String("s3Region", "", "S3 region")
+	s3Provider  = flag.String("s3Provider", "", "S3 provider")
+	s3Endpoint  = flag.String("s3Endpoint", "", "S3 endpoint")
+)
+
+const (
+	maxRetries     = 3
+	uuidLen        = 36
+	totalOrdered   = "TOTAL ORDERED"
+	partialOrdered = "PARTIAL ORDERED"
+	totalRandom    = "TOTAL RANDOM"
+	nullVal        = "\\N"
+)
+
+var (
+	faker    *gofakeit.Faker
+	orderMap = map[string]string{
+		"total ordered":   totalOrdered,
+		"partial ordered": partialOrdered,
+		"total random":    totalRandom,
+	}
+)
+
+// Initialize Faker instance
+func init() {
+	seed := time.Now().UnixNano()
+	faker = gofakeit.New(seed)
+	log.Printf("Faker seed: %d", seed)
+}
+
+// Column fields is read only
+type Column struct {
+	Name      string
+	Type      string
+	IsPK      bool
+	IsUnique  bool
+	Order     string
+	StdDev    float64
+	Mean      float64
+	MinLen    int
+	MaxLen    int
+	NullRatio int
+}
+
+func (c *Column) canThisValNull() bool {
+	if c.IsPK || c.NullRatio == 0 {
+		return false
+	} else if c.NullRatio == 100 {
+		return true
+	}
+	return faker.Number(1, 100) <= c.NullRatio
```

Comment:
@YangKeao Is here the tiny bug about null ratio?

Reply:
No. It's caused by my own modification. This PR is fine 🙇 .